package/firmware/ipq-wifi/Makefile

```diff
@@ -47,12 +47,16 @@ Don't install it for any other device!
   define Package/ipq-wifi-$(1)/install-overlay
 	$(INSTALL_DIR) $$(1)/lib/firmware/ath10k/QCA4019/hw1.0
 	$(INSTALL_DATA) ./$(2) $$(1)/lib/firmware/ath10k/QCA4019/hw1.0/board-2.bin
+	if [ -n "$(4)" ] ; then \
+		$(INSTALL_DIR) $$(1)/lib/firmware/ath10k/$(4) ;\
+		$(INSTALL_DATA) ./$(5) $$(1)/lib/firmware/ath10k/$(4)/board-2.bin ;\
+	fi
   endef
 
   PREV_BOARD+=ipq-wifi-$(1)
 endef
 
-#$(eval $(call generate-ipq-wifi-package,<devicename>,<filename>,<displayname>))
+#$(eval $(call generate-ipq-wifi-package,<devicename>,<filename>,<displayname>[,addl-path, addl-file]))
```

Comment:
hm, wouldn't it be better to extend the 

generate-ipq-wifi-package so it always takes the wifi-chip and board-2.bin path

and do customized helpers for the IPQ4019, QCA9888 (and QCA9984)

generate-ipq-wifi-package-ipq4019
generate-ipq-wifi-package-qca9888
(generate-ipq-wifi-package-qca9984 - no need yet, feel free to skip that)

Then you could have two packages for your EA8300. One for the QCA4019 board-2.bin 
and the other one for the QCA9888. The biggest hurdle would be to make it so that they
don't conflict with one another. But I think this is the cleanest way to do it, apart from
splitting the board-2.bin from the ath10k-firmware packages into separate packages.
What do you think?

Reply:
I think I left the biggest challenge for last 😉 

Not so much in a technical sense, but in a philosophy sense. 

I've got no concerns about needing to specify two (or even three) "packages" to get the proper board data into the build for a three-radio device. Only the device maintainers should ever have to worry about it, as they should be "default" for the device and just appear in the build.

I agree that there is benefit in making things modular and reusable. I haven't been able to convince myself that the ipq40xx platform is the only one that will be impacted. It wouldn't surprise me that as the performance demands on the radios increase, the need to specifically tune/tweak the parameters increases as well. In some sense, it could be "The new normal" that any new, mid- or upper-range device "needs" its custom data for at least as long as it takes for the files to make their way upstream and back again. The complexities of two radios on a single band just add to that feeling.

I'm still thinking what the advantage is, if any, of tying it to a platform, rather than having it available based on the use of a specific board and/or radio chip. I'm going to sleep on that one. 

Am I correct in assuming that the use of this approach is supposed to be only an interim thing until the properly "indexed" board-file segments are incorporated in the upstream sources? If so, then any "transition" package wouldn't be needed as newly ported devices could go with a new scheme, and the ones on an older scheme should naturally fall away as the segments are incorporated upstream.

Reply:
> I agree that there is benefit in making things modular and reusable. I haven't been able to convince myself that the ipq40xx platform is the only one that will be impacted. It wouldn't surprise me that as the performance demands on the radios increase, the need to specifically tune/tweak the parameters increases as well. In some sense, it could be "The new normal" that any new, mid- or upper-range device "needs" its custom data for at least as long as it takes for the files to make their way upstream and back again. The complexities of two radios on a single band just add to that feeling.

IPQ40XX is not the only platform, all wave v2 (and the older v1 customer products too) hardware is. You can extract the board-2.bin from Kvalo's ath10k-firmware repository and find tons of special boarddata for specific devices. As for OpenWrt. The main driving force behind the package was due to OpenMesh A42. However, somehow the boarddata files that @ecsv sent upstream were merged in a very timely fashion before this became a problem there. That said, Kalle does queue up the boarddata, you can find that he posted a mail of his latest queue on the mailing-list.
 
https://www.mail-archive.com/ath10k@lists.infradead.org/msg09966.html

> I'm still thinking what the advantage is, if any, of tying it to a platform, rather than having it available based on the use of a specific board and/or radio chip. I'm going to sleep on that one.

Yes, if you want you can make the ipq-wifi packages per platform. But you don't have to, personally I would wait around until one ath79 or ipq806x (upcoming ipq8074) user needs it. 

> Am I correct in assuming that the use of this approach is supposed to be only an interim thing until the properly "indexed" board-file segments are incorporated in the upstream sources? If so, then any "transition" package wouldn't be needed as newly ported devices could go with a new scheme, and the ones on an older scheme should naturally fall away as the segments are incorporated upstream.

Yes, we cleaned out the ipq-wifi package when the boarddata moved to upstream. See
25a72f5a01f80116d4cd0aafd99c0fd091be3fbd
e6bd568051c7ca77a59783fe50203f6e2a427f19
dceee8cc09a54f96dee82a6e95e3e77a6dd7ac13